### ISSUE TITLE
nativequery result do not map to entities when entity properties refers a associated entity

### ISSUE THREAD

**Issue by JoshiManjila** [Impetus/kundera#761]
Dear sir,

When we tested cassandra native queries using kundera  In our testing scenario, we need to convert the query result to Vaccation.class., one entity.

But the Vaccation class has property doctor which refers another entity (Many to One relation). In such cases, kundera em.resultList() returns Vacation List but with doctor  as null.

 List<Vacation> vacations =em.createNativeQuery(query, Vacation.class);

vacation.doctor is null

Why cant make kundera JPA create a doctor reference while the mapping to vaccation.class  going on.

Even if it don't support join query.,as doctor id returns in the native query refers to doctor object. and    if no fetching of doctor is be needed. then shouldn't it possible to return a list with  entity reference?

Please send your earliest replay.

Thanking you

Joshi.

**Comment by karthikprasad13** [comment]
Hi @JoshiManjila 

We have added the fix. You can build the source from trunk and start using this feature.

-Karthik

**Comment by JoshiManjila** [comment]
thank you verymuch Karthik.

**Comment by JoshiManjila** [comment]
Dear karthik,

when I build from source, I am getting error as follows

] kundera-cassandra-parent
[INFO] kundera-cassandra
[INFO] kundera-cassandra-pelops
[INFO] kundera-cassandra-ds-driver
[INFO]  
[INFO] ------------------------------------------------------------------------
[INFO] Building kundera-cassandra-parent 3.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] Downloading: http://mvn.riptano.com/content/repositories/public/com/impetus/kundera/core/kundera-core/3.0-SNAPSHOT/maven-metadata.xml
[WARNING] Could not transfer metadata com.impetus.kundera.core:kundera-core:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer com.impetus.kundera.core:kundera-core:3.0-SNAPSHOT/maven-metadata.xml from http://mvn.riptano.com/content/repositories/public was cached in the local repository, resolution will not be reattempted until the update interval of riptano has elapsed or updates are forced. Original error: Could not transfer metadata com.impetus.kundera.core:kundera-core:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[INFO] Downloading: http://mvn.riptano.com/content/repositories/public/com/impetus/kundera/core/kundera-core/3.0-SNAPSHOT/kundera-core-3.0-SNAPSHOT.pom
[WARNING] Failure to transfer com.impetus.kundera.core:kundera-core:3.0-SNAPSHOT/maven-metadata.xml from http://mvn.riptano.com/content/repositories/public was cached in the local repository, resolution will not be reattempted until the update interval of riptano has elapsed or updates are forced. Original error: Could not transfer metadata com.impetus.kundera.core:kundera-core:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[INFO] Downloading: http://mvn.riptano.com/content/repositories/public/com/impetus/kundera/core/fallback-impl/3.0-SNAPSHOT/maven-metadata.xml
[WARNING] Could not transfer metadata com.impetus.kundera.core:fallback-impl:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer com.impetus.kundera.core:fallback-impl:3.0-SNAPSHOT/maven-metadata.xml from http://mvn.riptano.com/content/repositories/public was cached in the local repository, resolution will not be reattempted until the update interval of riptano has elapsed or updates are forced. Original error: Could not transfer metadata com.impetus.kundera.core:fallback-impl:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[INFO] Downloading: http://mvn.riptano.com/content/repositories/public/com/impetus/kundera/core/fallback-impl/3.0-SNAPSHOT/fallback-impl-3.0-SNAPSHOT.pom
[INFO] Downloading: http://mvn.riptano.com/content/repositories/public/com/impetus/kundera/client/kundera-elastic-search/3.0-SNAPSHOT/maven-metadata.xml
[WARNING] Could not transfer metadata com.impetus.kundera.client:kundera-elastic-search:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer com.impetus.kundera.client:kundera-elastic-search:3.0-SNAPSHOT/maven-metadata.xml from http://mvn.riptano.com/content/repositories/public was cached in the local repository, resolution will not be reattempted until the update interval of riptano has elapsed or updates are forced. Original error: Could not transfer metadata com.impetus.kundera.client:kundera-elastic-search:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[INFO] Downloading: http://mvn.riptano.com/content/repositories/public/com/impetus/kundera/client/kundera-elastic-search/3.0-SNAPSHOT/kundera-elastic-search-3.0-SNAPSHOT.pom
[WARNING] Failure to transfer com.impetus.kundera.client:kundera-elastic-search:3.0-SNAPSHOT/maven-metadata.xml from http://mvn.riptano.com/content/repositories/public was cached in the local repository, resolution will not be reattempted until the update interval of riptano has elapsed or updates are forced. Original error: Could not transfer metadata com.impetus.kundera.client:kundera-elastic-search:3.0-SNAPSHOT/maven-metadata.xml from/to riptano (http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] kundera-cassandra-parent ........................... FAILURE [  4.072 s]
[INFO] kundera-cassandra .................................. SKIPPED
[INFO] kundera-cassandra-pelops ........................... SKIPPED
[INFO] kundera-cassandra-ds-driver ........................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE

Would you please help me build it successfully. I think there is some permission problem exist.

Thanking you

Joshi

**Comment by karthikprasad13** [comment]
Hi @JoshiManjila 

Sorry for replying so late. We tried to build Kundera in a fresh repository and did not face any such issue. Please check if you have any issue with your network connectivity.

-Karthik.

**Comment by JoshiManjila** [comment]
Dear Karthik,

When I am building kundera-core from source following tests are getting failed. I am building on jdk 8.

Would please help me at this point. I am still trying for building kundera- cassandra jar from source.

Joshi

(1) MetadataUtilsTest
Running com.impetus.kundera.metadata.MetadataUtilsTest
Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.386 sec <<< FAILURE!
testSerializeKeys(com.impetus.kundera.metadata.MetadataUtilsTest)  Time elapsed: 0.084 sec  <<< FAILURE!
junit.framework.ComparisonFailure: null expected:<key[3~key2~key1]> but was:<key[1~key2~key3]>
    at junit.framework.Assert.assertEquals(Assert.java:81)
    at junit.framework.Assert.assertEquals(Assert.java:87)
    at com.impetus.kundera.metadata.MetadataUtilsTest.testSerializeKeys(MetadataUtilsTest.java:120)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
    at org.junit.runners.BlockJUnit4ClassRunner.runNotIgnored(BlockJUnit4ClassRunner.java:79)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:71)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:49)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
    at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
    at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
    at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)

(2)  RoundRobinBalancingPolicyTest

Running com.impetus.kundera.service.policy.RoundRobinBalancingPolicyTest
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.018 sec <<< FAILURE!
testGetPool(com.impetus.kundera.service.policy.RoundRobinBalancingPolicyTest)  Time elapsed: 0.018 sec  <<< FAILURE!
junit.framework.AssertionFailedError: expected:java.lang.Object@6d6c7725 but was:java.lang.Object@37d13a24
    at junit.framework.Assert.fail(Assert.java:47)
    at junit.framework.Assert.failNotEquals(Assert.java:283)
    at junit.framework.Assert.assertEquals(Assert.java:64)
    at junit.framework.Assert.assertEquals(Assert.java:71)
    at com.impetus.kundera.service.policy.RoundRobinBalancingPolicyTest.testGetPool(RoundRobinBalancingPolicyTest.java:76)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
    at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
    at org.junit.runners.BlockJUnit4ClassRunner.runNotIgnored(BlockJUnit4ClassRunner.java:79)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:71)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:49)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
    at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
    at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
    at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)

**Comment by mevivs** [comment]
@JoshiManjila 

```
(http://mvn.riptano.com/content/repositories/public): Not authorized , ReasonPhrase:Unauthorized.
```

check this on log. Also, in case you want to build jar only! you may skip test as

```
mvn clean install -DskipTests
```

HTH,
-Vivek

**Comment by JoshiManjila** [comment]
Hi vivek,

Thank you very much for your reply. using mvn clean install -DskipTests, we could create jars. Let us check for the fix that you made.

Thanking you

Joshi

**Comment by chhavigangwal** [comment]
@JoshiManjila 

You can also use kundera-cassandra 3.0 to check this support.

Chhavi

**Comment by JoshiManjila** [comment]
Dear Chhavi

We tested by adding maven dependency for repository for 3.0. But this fix is not included

**Comment by chhavigangwal** [comment]
@JoshiManjila 

What exception are you getting now ?

Chhavi

**Comment by JoshiManjila** [comment]
Dear Chhavi,

In the above mentioned case(maven repository with version 3.0), when we execute a native query from kundera , like
em.createNativeQuery(query,Doctor.Tbl), after executing the query, mapped doctorTbl object , no foreign key references are mapped to the refered object   . ie any foreign-key reference mentioned in DoctorTbl are null in the mapped object.

Joshi

**Comment by chhavigangwal** [comment]
@JoshiManjila 

We are looking into the issue at our end again and will update you soon.

Chhavi

**Comment by JoshiManjila** [comment]
Dear Chhavi,

We tested here it with kundera trunk there also we are getting the same.

Joshi

**Comment by JoshiManjila** [comment]
Dear Chhavi,

If this is issue still there in version 3.0 please change the status of this issue to unresolved instead of fixed

Joshi

**Comment by chhavigangwal** [comment]
@JoshiManjila 

Please check , releasing the fix  with 3.1.

Chhavi
